stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/dialog/ModeratedMessageDialog.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.message.dialog
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.common.model.ModeratedMessageOption
+import io.getstream.chat.android.ui.common.internal.FullScreenDialogFragment
+import io.getstream.chat.android.ui.databinding.StreamUiDialogModeratedMessageBinding
+
+internal class ModeratedMessageDialog private constructor() : FullScreenDialogFragment() {
+
+    private var _binding: StreamUiDialogModeratedMessageBinding? = null
+    private val binding: StreamUiDialogModeratedMessageBinding get() = _binding!!
+
+    private lateinit var message: Message
+
+    private var selectionHandler: DialogSelectionHandler? = null
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = StreamUiDialogModeratedMessageBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        consumeMessageArg()
+        setupDismissalArea()
+        initSelectionListeners()
+    }
+
+    override fun onDestroyView() {
+        _binding = null
```

Comment:
We should most likely reset the `messageArg` in case the dialog is destroyed, before onViewCreated. maybe just in onDestroy/dismiss or something

Reply:
@TGazica actually - instead of having static message arg, could we just have the property in a bundle or something for the dialog? Or as a property in the class it self. That way we avoid having static data that jsut hangs up the memory potentially

Reply:
Can't go in the bundle, the Message class is neither parcelable or serializable. I can just pass the messageId but that would complicate the logic later down the pipeline. I built it in the same way as `MessageOptionsDialogFragment`. But i can set it so that it is a public var and set it once the dialog has been initialized.